platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformStateAccessorSingleton.java

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state;
+
+import static com.swirlds.platform.state.PlatformStateConverter.convert;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.platform.state.MinimumJudgeInfo;
+import com.hedera.hapi.platform.state.PlatformState;
+import com.hedera.hapi.platform.state.PlatformState.Builder;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import com.swirlds.platform.event.AncientMode;
+import com.swirlds.platform.state.schemas.V0540PlatformStateSchema;
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.state.merkle.StateMetadata;
+import com.swirlds.state.merkle.singleton.SingletonNode;
+import com.swirlds.state.spi.CommittableWritableStates;
+import com.swirlds.state.spi.ReadableSingletonState;
+import com.swirlds.state.spi.WritableSingletonState;
+import com.swirlds.state.spi.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class is a drop-in replacement for {@link PlatformState} that uses singleton object under the hood.`
+ * It assumes that the instance of {@link MerkleRoot} passed in the constructor is a {@link MerkleStateRoot} instance.
+ * It registers the singleton object for the platform state in the {@link MerkleStateRoot} if it is not already present.
+ */
+public class PlatformStateAccessorSingleton implements PlatformStateAccessor {
```

Comment:
given that the `get`s and `set`s on this class aren't simple `=` assignments, it might make sense to verify for each one whether a `get` after a `set` yields the expected value